### ISSUE TITLE
Multiple ANR with error "Input dispatching timed out"

### ISSUE THREAD

**Issue by WelliRigo** [nizwar/screen_capture_event#28]
I have multiple ANR errors with the following message:

id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.getLastModified
Input dispatching timed out

It happens only at Android 11 (SDK 30)

```
  #00  pc 0x000000000009b088  /apex/com.android.runtime/lib64/bionic/libc.so (__getdents64+8)
  #01  pc 0x0000000000050a0c  /apex/com.android.runtime/lib64/bionic/libc.so (readdir+100)
  #02  pc 0x000000000002138c  /apex/com.android.art/lib64/libopenjdk.so (Java_java_io_UnixFileSystem_list0+180)
  at java.io.UnixFileSystem.list0 (Native method)
  at java.io.UnixFileSystem.list (UnixFileSystem.java:346)
  at java.io.File.list (File.java:1131)
  at java.io.File.listFiles (File.java:1295)
  at id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.getLastModified (ScreenCaptureEventPlugin.java:231)
  at id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.updateScreenRecordStatus (ScreenCaptureEventPlugin.java:164)
  at id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.onMethodCall (ScreenCaptureEventPlugin.java:76)
  at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage (MethodChannel.java:258)
  at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler (DartMessenger.java:295)
```

**Comment by seokhyeonJN** [comment]
any update?

**Comment by WelliRigo** [comment]
Still not, I had to create my own screencapture detector

**Comment by seokhyeonJN** [comment]
@WelliRigo would you tell me if you used channel method?

**Comment by WelliRigo** [comment]
Yes, because the implementation is in Java, similar to https://github.com/nizwar/screen_capture_event/blob/master/android/src/main/java/id/nizwar/screen_capture_event/ScreenCaptureEventPlugin.java

**Comment by owenmoon1** [comment]
Any update?

**Comment by JDongKhan** [comment]
Any update?

**Comment by emreerkaslan** [comment]
This issue is still valid and causing a lot of ANRs, appreciate if you can prioritize.

Attaching a stacktrace here:

libopenjdk.so
Java_java_io_UnixFileSystem_list0 + 180
java.io.UnixFileSystem.list0 (Native method)
java.io.UnixFileSystem.list (UnixFileSystem.java:402)
java.io.File.list (File.java:1145)
java.io.File.listFiles (File.java:1230)
id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.getLastModified (ScreenCaptureEventPlugin.java:260)
id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.updateScreenRecordStatus (ScreenCaptureEventPlugin.java:177)
id.nizwar.screen_capture_event.ScreenCaptureEventPlugin.onMethodCall (ScreenCaptureEventPlugin.java:79)
io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage (MethodChannel.java:267)
io.flutter.embedding.engine.dart.DartMessenger.invokeHandler (DartMessenger.java:292)
io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0 (DartMessenger.java:319)
android.os.Handler.handleCallback (Handler.java:958)
android.os.Handler.dispatchMessage (Handler.java:99)
android.os.Looper.loopOnce (Looper.java:205)
android.os.Looper.loop (Looper.java:294)
android.app.ActivityThread.main (ActivityThread.java:8391)
java.lang.reflect.Method.invoke (Native method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:640)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:995)

------

[✓] Flutter (Channel stable, 3.38.3, on macOS 15.6.1 24G90 darwin-arm64, locale en-TR) [513ms]
    • Flutter version 3.38.3 on channel stable ....
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 19074d12f7 (5 weeks ago), 2025-11-20 17:53:13 -0500
    • Engine revision 13e658725d
    • Dart version 3.10.1
    • DevTools version 2.51.1
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, omit-legacy-version-file,
      enable-lldb-debugging

[✓] Android toolchain - develop for Android devices (Android SDK version 36.1.0) [4.3s]
    • Android SDK at .....
    • Emulator version 36.1.9.0 (build_id 13823996) (CL:N/A)
    • Platform android-36, build-tools 36.1.0
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home/bin/java
      This JDK is specified in your Flutter configuration.
      To change the current JDK, run: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version Java(TM) SE Runtime Environment (build 17.0.12+8-LTS-286)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.4) [1,923ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16F6
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [6ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Connected device (3 available) [9.9s]
    • macOS (desktop)                   • macos                     • darwin-arm64   • macOS 15.6.1 24G90 darwin-arm64
    • Chrome (web)                      • chrome                    • web-javascript • Google Chrome 143.0.7499.170

[✓] Network resources [477ms]
    • All expected network resources are available.

• No issues found!
